File: gravity_simulation.f90
!gravity_simulation

program gravity_simulation
    implicit none
    
    !declare Variables
    real :: m1, m2, r, force
    real, parameter :: G = 6.674e-11

    !Initialize Variables
    m1 = 1.989e30
    m2 = 5.972e24
    r = 1.496e11

    !Compute gravitational force
    force = gravitational_force(m1, m2, r)

    !Print result
    call print_result(m1, m2, r, force)

contains
    
    !Function to compute gravitational force
    function gravitational_force(m1, m2, r) result(force)
        real, intent(in) :: m1, m2, r
        real :: force
        force = G * m1 * m2 / r**2
    end function gravitational_force

    !Subroutine to print result
    Subroutine print_result(m1, m2, r, force)
        real, intent(in) :: m1, m2, r, force

        print *, 'Mass of the Sun', m1, 'kg'
        print *, 'Mass of the Earth', m2, 'kg'
        print *, 'Distance', r, 'm'
        print *, 'Gravitational force = ', force, 'N'
    end Subroutine print_result

end program gravity_simulation

File: gravity_simulation.py
# gravity_simulation.py
# Demonstrates functions and subroutines in Python

# Constants
G = 6.67430e-11  # Gravitational constant (m^3 kg^-1 s^-2)

# Function to compute gravitational force
def gravitational_force(m1, m2, r):
    return G * m1 * m2 / r**2

# Subroutine to print results
def print_results(m1, m2, r, force):
    print(f"Mass 1 (Sun): {m1} kg")
    print(f"Mass 2 (Earth): {m2} kg")
    print(f"Distance: {r} m")
    print(f"Gravitational Force: {force} N")

# Main program
if __name__ == "__main__":
    # Initialize masses and distance
    m1 = 1.989e30  # Mass of the Sun (kg)
    m2 = 5.972e24  # Mass of the Earth (kg)
    r = 1.496e11   # Distance between Sun and Earth (m)

    # Compute gravitational force
    force = gravitational_force(m1, m2, r)

    # Print the results
    print_results(m1, m2, r, force)